tests/apps/qchem/test_dynamics.py

```diff
@@ -326,3 +332,137 @@ def test_negative_state(self, e):
         with pytest.raises(ValueError, match="The excited state must not contain negative values"):
             samples, state, _ = e
             dynamics.prob(samples, [-i for i in state])
+
+
+@pytest.mark.parametrize("c", [c1, c2])
+class TestSampleCoherent:
+    """Tests for the function ``strawberryfields.apps.qchem.dynamics.sample_coherent``"""
+
+    def test_invalid_n_samples(self, c):
+        """Test if function raises a ``ValueError`` when a number of samples less than one is
+        requested."""
+        with pytest.raises(ValueError, match="Number of samples must be at least one"):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, U, w, 0)
+
+    def test_invalid_time(self, c):
+        """Test if function raises a ``ValueError`` when a negative time is given."""
+        with pytest.raises(ValueError, match="Time must be zero or positive"):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, -t, U, w, ns)
+
+    def test_negative_frequency(self, c):
+        """Test if function raises a ``ValueError`` when negative frequencies are given."""
+        with pytest.raises(ValueError, match="Vibrational frequencies must be larger than zero"):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, U, -w, ns)
+
+    def test_zero_frequency(self, c):
+        """Test if function raises a ``ValueError`` when zero frequencies are given."""
+        with pytest.raises(ValueError, match="Vibrational frequencies must be larger than zero"):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, U, 0.0 * w, ns)
+
+    def test_complex_unitary(self, c):
+        """Test if function raises a ``ValueError`` when a complex unitary is given."""
+        with pytest.raises(
+            ValueError, match="The normal mode to local mode transformation matrix must be real"
+        ):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, 1.0j * U, w, ns)
+
+    def test_invalid_loss(self, c):
+        """Test if function raises a ``ValueError`` when the loss parameter is specified outside
+        of range."""
+        with pytest.raises(
+            ValueError, match="Loss parameter must take a value between zero and one"
+        ):
+            alpha, t, U, w, ns = c
+            dynamics.sample_coherent(alpha, t, U, w, ns, -1)
```

Comment:
Note that the tests are failing because we are testing here for an invalid loss but we do not raise the above value error anymore.